client/src/www/app/outline_server_repository/access_key_serialization.ts

```diff
@@ -40,7 +40,7 @@ function parseShadowsocksSessionConfigJson(responseBody: string): ShadowsocksSes
   const responseJson = JSON.parse(responseBody);
 
   if ('error' in responseJson) {
-    throw new errors.SessionConfigError(responseJson.error.message);
+    throw new errors.SessionConfigError(responseJson.error.message, { cause: responseJson.error.cause });
```

Comment:
Let's name it 'details' instead of 'cause', since that's the product language.

```suggestion
    throw new errors.SessionConfigError(responseJson.error.message, { cause: responseJson.error.details });
```

Reply:
`cause` is the name the js standard uses. We can't change that, and it makes sense anyway.

What perhaps is confusing is putting the error details in the cause field.
Can we add a details field to `SessionConfigError` instead of setting the `cause`?

Reply:
Shouldn't this be the error itself, not the details on the error? `cause: responseJson.error`?

Reply:
I think one problem with the design is that we are overloading SessionConfigError for fetch/parse errors and the server provider errors. In a way the provider error is the cause of the SessionConfigError, but we want to show the provider error instead.

I propose we align the error type to the message we show in the error. If we show a different message, the error is important enough to be defined explicitly.

With that in mind, define perhaps a `SessionProviderError`. This has its own semantics, with message and details (not cause!), so the error must have those fields. We don't have a cause, so don't set it.

In the error handling, you can check `error.details` to set up the link. Then don't use showErrorDetailDialog. That's for a different use case. Just open an alert with the error details.